x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/GetApiKeyResponse.java

```diff
@@ -29,47 +29,75 @@
  */
 public final class GetApiKeyResponse extends ActionResponse implements ToXContentObject {
 
-    private final ApiKey[] foundApiKeysInfo;
+    public static final GetApiKeyResponse EMPTY = new GetApiKeyResponse(List.of());
 
-    public GetApiKeyResponse(Collection<ApiKey> foundApiKeysInfo) {
-        Objects.requireNonNull(foundApiKeysInfo, "found_api_keys_info must be provided");
-        this.foundApiKeysInfo = foundApiKeysInfo.toArray(new ApiKey[0]);
-    }
+    private final List<Item> foundApiKeyInfoList;
 
-    public static GetApiKeyResponse emptyResponse() {
-        return new GetApiKeyResponse(List.of());
+    public GetApiKeyResponse(Collection<ApiKey> foundApiKeyInfos) {
```

Comment:
Wondering if we also want to move the constructor arg to a `List`. Ultimately, we're dealing with a collection where order matters which is better capture by `List`

Reply:
I've thought of that too. The usage of a `Collection` instead of a `List` goes back to https://github.com/elastic/elasticsearch/blob/43b8ca00865c2846346f01d834075432e1362bb6/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/ScrollHelper.java#L38
At that point, the indexed access in the result, afforded by a `List` compared to a `Collection`, does not look important or relevant. Even in the transport response objects (`GetApiKeyResponse`), indexed access is only useful in tests for simplicity. So I thought changing it to `List`s more broadly, was too large of a code change compared to the motivation.